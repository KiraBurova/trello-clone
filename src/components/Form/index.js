import React from 'react';

import { Container, Input, Button, Title, ErrorText, LoadingText } from './styles/form';

const Form = ({ formTitle, buttonTitle, handleSumbit, handleOnChange, errorText, loading, isValid }) => {
  return (
    <Container onSubmit={handleSumbit}>
      <Title>{formTitle}</Title>
      <Input placeholder='Email' name='email' onChange={handleOnChange}></Input>
      <Input placeholder='Password' name='password' type='password' onChange={handleOnChange}></Input>
      <Button disabled={!isValid}>{buttonTitle}</Button>
      <ErrorText>{errorText}</ErrorText>
      <LoadingText>{loading}</LoadingText>
    </Container>
  );
};

export default Form;
