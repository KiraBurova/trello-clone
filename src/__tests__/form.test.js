import React from 'react';

import { render } from '@testing-library/react';

import Form from '../components/Form';

describe('<Form />', () => {
  it('renders', () => {
    const formData = {
      formTitle: 'Sign up form',
      buttonTitle: 'Sign up',
    };
    const { container, getByPlaceholderText, getByText } = render(<Form {...formData} />);

    expect(getByPlaceholderText('Password')).toBeTruthy();
    expect(getByPlaceholderText('Email')).toBeTruthy();
    expect(getByText(formData.buttonTitle)).toBeTruthy();
    expect(getByText(formData.formTitle)).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders with disabled button', () => {
    const formData = {
      buttonTitle: 'Sign up',
      isValid: false,
    };
    const { container, getByText } = render(<Form {...formData} />);

    expect(getByText(formData.buttonTitle).disabled).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders with error', () => {
    const formData = {
      errorText: 'Error',
    };
    const { container, getByText } = render(<Form {...formData} />);

    expect(getByText(formData.errorText)).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders with loading text', () => {
    const formData = {
      loading: 'Loading!',
    };
    const { container, getByText } = render(<Form {...formData} />);

    expect(getByText(formData.loading)).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
