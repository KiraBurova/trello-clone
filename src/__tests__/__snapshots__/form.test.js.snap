// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders 1`] = `
<form
  class="form__Base-sc-1grfi8t-0 hLGYyj"
>
  <h2
    class="form__Title-sc-1grfi8t-1 ekfsMg"
  >
    Sign up form
  </h2>
  <input
    class="form__Input-sc-1grfi8t-2 czvKJK"
    name="email"
    placeholder="Email"
  />
  <input
    class="form__Input-sc-1grfi8t-2 czvKJK"
    name="password"
    placeholder="Password"
    type="password"
  />
  <button
    class="form__Button-sc-1grfi8t-3 ijFtjC"
    disabled=""
  >
    Sign up
  </button>
  <p
    class="form__ErrorText-sc-1grfi8t-4 ioXfqK"
  />
  <p
    class="form__LoadingText-sc-1grfi8t-5 hVQvTy"
  />
</form>
`;

exports[`<Form /> renders with disabled button 1`] = `
<form
  class="form__Base-sc-1grfi8t-0 hLGYyj"
>
  <h2
    class="form__Title-sc-1grfi8t-1 ekfsMg"
  />
  <input
    class="form__Input-sc-1grfi8t-2 czvKJK"
    name="email"
    placeholder="Email"
  />
  <input
    class="form__Input-sc-1grfi8t-2 czvKJK"
    name="password"
    placeholder="Password"
    type="password"
  />
  <button
    class="form__Button-sc-1grfi8t-3 ijFtjC"
    disabled=""
  >
    Sign up
  </button>
  <p
    class="form__ErrorText-sc-1grfi8t-4 ioXfqK"
  />
  <p
    class="form__LoadingText-sc-1grfi8t-5 hVQvTy"
  />
</form>
`;

exports[`<Form /> renders with error 1`] = `
<form
  class="form__Base-sc-1grfi8t-0 hLGYyj"
>
  <h2
    class="form__Title-sc-1grfi8t-1 ekfsMg"
  />
  <input
    class="form__Input-sc-1grfi8t-2 czvKJK"
    name="email"
    placeholder="Email"
  />
  <input
    class="form__Input-sc-1grfi8t-2 czvKJK"
    name="password"
    placeholder="Password"
    type="password"
  />
  <button
    class="form__Button-sc-1grfi8t-3 ijFtjC"
    disabled=""
  />
  <p
    class="form__ErrorText-sc-1grfi8t-4 ioXfqK"
  >
    Error
  </p>
  <p
    class="form__LoadingText-sc-1grfi8t-5 hVQvTy"
  />
</form>
`;

exports[`<Form /> renders with loading text 1`] = `
<form
  class="form__Base-sc-1grfi8t-0 hLGYyj"
>
  <h2
    class="form__Title-sc-1grfi8t-1 ekfsMg"
  />
  <input
    class="form__Input-sc-1grfi8t-2 czvKJK"
    name="email"
    placeholder="Email"
  />
  <input
    class="form__Input-sc-1grfi8t-2 czvKJK"
    name="password"
    placeholder="Password"
    type="password"
  />
  <button
    class="form__Button-sc-1grfi8t-3 ijFtjC"
    disabled=""
  />
  <p
    class="form__ErrorText-sc-1grfi8t-4 ioXfqK"
  />
  <p
    class="form__LoadingText-sc-1grfi8t-5 hVQvTy"
  >
    Loading!
  </p>
</form>
`;
